carbon-design-system/carbon
# chore: add decorator in dataTable

Closes #18005 

Add decorator in dataTable component

#### Changelog

Added a new component `TableDecoratorRow`
Added a new deprecated warning function for components

**Changed**

1. marked `slug` as deprecated prop in all dataTable components
2. marked `TableSlugRow` component as deprecated
3. Added few decorators in storybook as` ai-lable` and `decorator` both.

#### Testing / Reviewing

1. Go to Storybook > DataTable > AILabel story
2. Check each story under that, All AI slug stories are now a mix of slug and decorator, so all should look fine.
3. Added few examples of decorator Tooltip as well, that too should look fine.
